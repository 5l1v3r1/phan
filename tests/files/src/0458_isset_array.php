<?php
call_user_func(
    /**
     * @param array{key:string} $data
     * @param string $str
     */
    function($data, $str) {
        if (isset($data['otherKey'])) {
            echo strlen($data);
        } else {
            echo strlen($data);
        }
        // It's possible that the phpdoc types were wrong. Don't warn (this isn't strict mode).
        if (isset($str[5])) {
            echo count($str);  '@phan-debug-var $str';
        }
        if (isset($str['field'])) {
            echo strlen($str);  // $str is definitely not a string
        }
    },
    [],
    'stringVar'
);
call_user_func(
    /**
     * @param array{0:?string,1:int} $data
     */
    function($data, $str) {
        if (isset($data['key'])) {
            echo strlen($data);
        } else {
            echo strlen($data);
        }
    }
);

call_user_func(
    /**
     * @param array{0:?string,1:int} $data
     */
    function($data) {
        if (isset($data[0])) {
            echo strlen($data);
        } else {
            echo strlen($data);
        }
    },
    ['x', 2]
);
