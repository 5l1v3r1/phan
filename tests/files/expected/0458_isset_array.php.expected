%s:9 PhanTypeMismatchArgumentInternal Argument 1 ($string) is array{otherKey:mixed,key:string} but \strlen() takes string
%s:11 PhanTypeMismatchArgumentInternal Argument 1 ($string) is array{key:string} but \strlen() takes string
%s:15 PhanDebugAnnotation @phan-debug-var requested for variable $str - it has union type non-empty-array<mixed,mixed>|object|string(real=non-empty-array<mixed,mixed>|object|string)
%s:18 PhanTypeMismatchArgumentInternalProbablyReal Argument 1 ($string) is non-empty-array<mixed,mixed>|object but \strlen() takes string
%s:21 PhanTypeMismatchArgument Argument 1 ($data) is array{} but Closure($data, $str) takes array{key:string} defined at %s:7
%s:24 PhanParamTooFewCallable Call with 0 arg(s) to Closure($data, $str) (as a provided callable) which requires 2 arg(s) defined at %s:28
%s:30 PhanTypeMismatchArgumentInternal Argument 1 ($string) is array{key:mixed,0:?string,1:int} but \strlen() takes string
%s:32 PhanTypeMismatchArgumentInternal Argument 1 ($string) is array{0:?string,1:int} but \strlen() takes string
%s:43 PhanTypeMismatchArgumentInternal Argument 1 ($string) is array{0:string,1:int} but \strlen() takes string
%s:45 PhanTypeMismatchArgumentInternal Argument 1 ($string) is array{0?:null,1:int} but \strlen() takes string
